### DISCUSSION TITLE
Possible to play a list of files in loop forever?

### DISCUSSION THREAD

**Discussion by Barabba11** [bluenviron/mediamtx#1059] [Questions]
Hi, the goal is to serve different TVs connected by ethernet, play all the files contained in the folder and repeat when it's finished. I suppose it has no sense on LAN, but multicast support can be interesting too.
Can this program do this? Thank you

**Comment by aler9** [comment]
Hello, you can do this by using ffmpeg playlists:

https://stackoverflow.com/questions/61302245/trying-to-loop-a-playlist-in-folder-with-ffmpeg

Use the concat filter and then publish to the server by using

```
-f rtsp rtsp://127.0.0.1:8554/stream
```

**Comment by Barabba11** [comment]
thank you, really appreciated!
I'm looking for a solution for windows, looks I've to adapt the bash to a dos-bat file, but I have no competence to build that.
There is any software in this world which is able to stream a playlist of videos? It looks a bit strange. I've found only something ultra-professional for TV broadcasters, I just need to stream without reencoding..
Thanks for all your ideas!

**Reply by aler9** [reply]
Just create a file like this, called `playlist.txt`

```
file1.mp4
file2.mp4
```

and call `ffmpeg`:

```
ffmpeg -f concat -safe 0 -stream_loop -1 \
-i  playlist.txt \
-c copy \
-f rtsp \
rtsp://127.0.0.1:8554/mystream
```

**Comment by Barabba11** [comment]
Thanks, I've tried different solution to try to solve myslef the problem, I can't start ffmpeg on listening mode (it opens a tcp port listening) for connection, viceversa it tries to connect to a server to the IP address specified, I've tried the ?listen or ?listen=1 option at the end an some others, no way. How can I do it?

to your code I had to make some corections but it's working :) thanks, I post it here:
playlist should contain;
```
file 'file1.mp4'
file 'file2.mp4'
```

command line I'm trying:
```
`ffmpeg -re -f concat -safe 0 -stream_loop -1 -i  playlist.txt -c copy -f  rtsp -rtsp_flags prefer_tcp -rtsp_transport tcp rtsp://192.168.0.100:2554/live.sdp?listen=1
```

**Comment by Barabba11** [comment]
After many attempts I guess the problem is not opening the port, but play the video itself. With TCP output it works, it creates the TCP port to listen, when VLC it connects it starts streaming (and stop when closed connection)
ffmpeg -re -f concat -safe 0 -stream_loop -1 -i  playlist.txt -c copy -f mpegts tcp://127.0.0.1:8888?listen 
WIth RTSP no way.. it tries to connect to that port, but not host it as server. Probably ffmpeg doesn't support rtsp streaming, now I try with rtmp

**Comment by Barabba11** [comment]
also with rtmp, udp and multicast it dowsn't work, buceause in thise ways it can't wait for an input to start the stream. So how can I force to "play"?

**Reply by aler9** [reply]
In order to publish a stream to the server, the right command is exclusively the one provided in the README, i.e.

```
ffmpeg .. -c copy -f rtsp rtsp://127.0.0.1:8554/stream
```

ffmpeg acts as a client to the server. The server is then responsible of serving the stream to any other client that wants to play it.

All the other commands listed here are used to create an autonomous RTSP or TCP/MPEGTS server, and there's no need to create one since you're already using rtsp-simple-server.

**Comment by Barabba11** [comment]
thank you for your kind reply, 
it looks for me a bit strange that with tcp ffmpeg is ready to serve VLC when it target the listening port, but with rtsp there aren't listening ports and it needs am external server/proxy to do that. If it is and ffmpeg is not able to servean rtsp like this please confirm me, and I will accept it, thanks.

A question about your program, is it possible to exclude completely UDP protocol form it? not listen fo incoming requests and don't do them by UDP? Unfortunately this protocol always produce glitches, even if the bandwith is more than necessary. Thanks

EDIT: last question, are you interested on produce something for DLNA? There are servers around but with poor results with incoming streams, I've tried Serviio, Universal Media Server, Simple DLNA (doesn't support input streams), poor results. I would try miniDLNA but it's for linux

**Reply by aler9** [reply]
> it looks for me a bit strange that with tcp ffmpeg is ready to serve VLC when it target the listening port, but with rtsp there aren't listening ports

They are two different protocols, TCP is way simpler than RTSP but lacks a lot of features, anyway i'm not a VLC developer, probably VLC doesn't have an embedded RTSP server, but i can't answer on that.

> A question about your program, is it possible to exclude completely UDP protocol form it

yes, set the `protocols` parameter in the configuration file:

```yml
protocols: [tcp]
```

> are you interested on produce something for DLNA?

Not at the moment, sorry, but since this project is open source, anyone is free to contribute.

**Reply by Barabba11** [reply]
Thans for reply, fo you have an idea how can I solve my issue? Ok I can use ffmpeg to "play" a playlist.. but I don't understand its methid to connect to an rtsp server to transmit the stream (it doesn't itself create a listening port, but it pretends to connect to something, I've sniffed it). So how your rtsp server can help me with ffmpeg? Can it accept somehow the strange ffmpeg request and serve all future rtsp requests from real clients? Thank you for your kind help

[a comment by an automated account is not shown]
